### ISSUE TITLE
Switch to Train/EvalResult

### ISSUE THREAD

**Issue by Erlemar** [Erlemar/pytorch_tempest#5]
Train/EvalResult seems promising, but has some bugs. When everything if fixed, I should switch to it.

**Comment by roma-goodok** [comment]
Hi, could you clarify what you mean by "switch to Train/EvalResult"?

**Comment by Erlemar** [comment]
In 0.9 Pytorch Lightning new objects were added - TrainResult and EvalResul  https://medium.com/pytorch/pytorch-lightning-0-9-synced-batchnorm-datamodules-and-final-api-aae885398a9d
It should have replaced logging with dictionaries, like I do here: https://github.com/Erlemar/pytorch_tempest/blob/master/src/lightning_classes/lightning_image_classification.py#L49

I tried that API and had some bugs, so I made an issue to switch to that API in future.

In 1.0 that API was removes, instead it is possible to use `self.log` method: https://pytorch-lightning.readthedocs.io/en/latest/lightning_module.html#training-epoch-level-metrics

So this issue now is about switching to `self.log` API instead of using dictionaries.

**Comment by Erlemar** [comment]
Switched to `self.log`.
